pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -279,7 +280,13 @@ func (r *Builder) mapMacStaticIps(vm *model.VM) (ipMap string, err error) {
 				}
 			}
 			dnsString := strings.Join(guestNetwork.DNS, ",")
-			configurations = append(configurations, fmt.Sprintf("%s:ip:%s,%s,%d,%s", guestNetwork.MAC, guestNetwork.IP, gateway, guestNetwork.PrefixLength, dnsString))
+
+			// if DNS is "", don't add it
+			if dnsString == "" {
+				configurations = append(configurations, fmt.Sprintf("%s:ip:%s,%s,%d", guestNetwork.MAC, guestNetwork.IP, gateway, guestNetwork.PrefixLength))
```

Comment:
Do we need it twice? if the DNS string is empty wouldn't it simply omitted in the end? if not how about extracting it to a func to reduce code duplication.

Reply:
the template is:
```go
fmt.Sprintf("%s:ip:%s,%s,%d,%s", guestNetwork.MAC, guestNetwork.IP, gateway, guestNetwork.PrefixLength, dnsString)
```

so if `dnsString == ""` we get a string that looks like `00:50:56:83:25:47:ip:172.29.3.193,172.29.3.1,16,`

virt-v2v does not like `,` at the end of the `--mac` command line arg, we can also clean it up by
`str = strings.TrimSuffix(str, ",")
`
not sure what is nicer ?

Reply:
thanks for clarifying, I personally find the trim nicer, but its really a meter of style so i'm fine with both.

Reply:
changed to use trim :+1: